### ISSUE TITLE
MAPPO RNN critic uses inconsistent batching (batchify vs jnp.repeat), causing incorrect hidden-state resets and broken GAE

### ISSUE THREAD

**Issue by jh2525** [FLAIROx/JaxMARL#169]
While implementing MAPPO with RNNs in JaxMARL, I found a critical batching mismatch between the actor and the centralized critic that leads to incorrect RNN hidden-state resets and invalid GAE computation.

Specifically, actor inputs (obs, done) are batched using JaxMARL's `batchify`, while the critic's global state is constructed using `jnp.repeat`. However, the critic still receives the actor-style last_done mask, which assumes `batchify` ordering. This causes a misalignment between the critic input and its RNN reset mask.

`batchify` stacks data in the following order:
```
[env1-agent1, env2-agent1, ..., envN-agent1,
 env1-agent2, env2-agent2, ..., envN-agent2, ...]
```

In the MAPPO critic, the global state is constructed as follows:
```python
world_state = last_obs["world_state"]  # (NUM_ENVS, state_dim)
world_state = jnp.repeat(
    world_state, env.num_agents, axis=0
)  # (NUM_ACTORS, state_dim)

```
This produces the ordering:
```
[env1, env1, env2, env2, env3, env3, ...]
```
However, the critic RNN input still uses the actor-style last_done mask:

```python
cr_in = (
    world_state[None, :],
    last_done[None, :],  # batched with `batchify`
)
```

As a result, the critic's hidden state is reset using a done signal that corresponds to a different (env, agent) pair than the one implied by the global state input. This causes the critic RNN to reset hidden states for incorrect trajectories, which in turn breaks temporal consistency and leads to invalid GAE computation.

**Comment by jh2525** [comment]
This issue was caused by a local modification in my implementation. The original JaxMARL, MAPPO code does not have this inconsistency. Sorry for the confusion, and thanks for the great library.
